nemo_aligner/data/nlp/builders.py

```diff
@@ -486,6 +503,17 @@ def build_dataloader(
         "pad_samples_to_global_batch_size": pad_samples_to_global_batch_size,
     }
 
+    if (
+        limit_train_batches is None
+        or (isinstance(limit_train_batches, float) and limit_train_batches > 1.0)
+        or (limit_train_batches <= 0)
+    ):
+        limit_train_batches = 1.0
+    if isinstance(limit_train_batches, float):
+        common_params["total_samples"] = int(limit_train_batches * len(dataset))
+    elif isinstance(limit_train_batches, int):
+        common_params["total_samples"] = min(limit_train_batches * gbs, len(dataset))
```

Comment:
Ensuring bad config doesn't get ignored:
```suggestion
    elif isinstance(limit_train_batches, int):
        common_params["total_samples"] = min(limit_train_batches * gbs, len(dataset))
    else:
        raise TypeError(type(limit_train_batches))
```